### PR TITLE
fix: toolbar ui

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes mf-4093

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx [hunk 1]
```diff
@@ -44,20 +49,22 @@ export function ToolboxHintAtTwitter(props: { category: 'wallet' | 'application'
             </Text>
         )
     }, [buttonStyle.iconSize, textMarginLeft])
-
-    const [mini, setMini] = useState(false)
+    const _mini = useMediaQuery(`(max-width: ${twitterBreakPoint}px)`)
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx]:
```suggestion
    const _mini = useMediaQuery(`(max-width: ${twitterBreakpoint}px)`)
```

File: packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+import { styled, ListItemButton, Typography, ListItemIcon, Box, useMediaQuery } from '@mui/material'
 import GuideStep from '../../../components/GuideStep/index.js'
 import { useI18N } from '../../../utils/index.js'
 import { useThemeSettings } from '../../../components/DataSource/useActivatedUI.js'
 import { ButtonStyle } from '../constant.js'
 import { useEffect, useMemo, useState } from 'react'
-import { MutationObserverWatcher } from '@dimensiondev/holoflows-kit'
 import { searchHomeLinkName } from '../utils/selector.js'
 
+const twitterBreakPoint = 1265
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx]: Where does this number come from?

File: packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx [hunk 3]
```diff
@@ -44,20 +49,22 @@ export function ToolboxHintAtTwitter(props: { category: 'wallet' | 'application'
...
-            { childList: true, subtree: true },
-            abortController.signal,
-        )
-        watch.addListener('onAdd', () => setMini(false), { signal: abortController.signal })
-        watch.addListener('onRemove', () => setMini(true), { signal: abortController.signal })
+        const observer = new MutationObserver((mutations) => {
+            setMini(!searchHomeLinkName().querySelector('span').evaluate())
+        })
+
+        observer.observe(searchHomeLinkName().evaluate()!, {
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx:60]: Need to unsubscribe.

File: packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx [hunk 4]
```diff
@@ -44,20 +49,22 @@ export function ToolboxHintAtTwitter(props: { category: 'wallet' | 'application'
             </Text>
         )
     }, [buttonStyle.iconSize, textMarginLeft])
-
-    const [mini, setMini] = useState(false)
+    const _mini = useMediaQuery(`(max-width: ${twitterBreakPoint}px)`)
+    const [mini, setMini] = useState(_mini)
 
     useEffect(() => {
         const abortController = new AbortController()
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx]: It didn't use.

File: packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx (hunk 2 above)
Comment (lelenei) [reply packages/mask/src/social-network-adaptor/twitter.com/injection/ToolboxHint_UI.tsx]: revert to original code

**Review by guanbinrui — DISMISSED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**